packages/react-charting/UnitTests/HeatMapChartUT.test.tsx

```diff
@@ -0,0 +1,676 @@
+    expect(result).toBeDefined();
+    expect(result).toEqual('1');
+    const result1 = instance._getOpacity(HeatMapDataNumaricPoints[1].legend);
+    expect(result1).toBeDefined();
+    expect(result1).toEqual('1');
+  });
+});
+
+runTest('_createLegendBars', () => {
+  test('Should return proper legends data', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    instance._colorScale = instance._getColorScale();
+    const result = instance._createLegendBars();
+    expect(result).toBeDefined();
+    expect(result.props.legends).toHaveLength(2);
+    const legends = result.props.legends;
+    expect(legends[0].title).toEqual('Execllent (0-200)');
+    expect(legends[1].title).toEqual('poor(200-400)');
+  });
+});
+
+runTest('_getColorScale', () => {
+  test('Should return proper color scale data', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    const colorScale = instance._getColorScale();
+    expect(colorScale).toBeDefined();
+    expect(colorScale(HeatMapDataNumaricPoints[0].value)).toEqual('rgb(144, 180, 215)');
+    expect(colorScale(HeatMapDataNumaricPoints[1].value)).toEqual('rgb(87, 108, 185)');
+    expect(colorScale(-100)).toEqual('rgb(202, 252, 245)');
+    expect(colorScale(700)).toEqual('rgb(0, 0, 124)');
+  });
+});
+
+runTest('_getXIndex', () => {
+  test('Should return proper xIndex for string xPoint', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    const xIndex = instance._getXIndex(xPoint[0]);
+    expect(xIndex).toEqual('Test1');
+  });
+
+  test('Should return proper xIndex for numeric xPoint', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    const xIndex = instance._getXIndex(xPoint1[0]);
+    expect(xIndex).toEqual('20000');
+  });
+
+  test('Should return proper xIndex for date xPoint', () => {
+    const instance = new HeatMapChartBase({
+      data: HeatMapDataNumaricPoints,
+      domainValuesForColorScale: domainValuesForColorScale,
+      rangeValuesForColorScale: rangeValuesForColorScale,
+    });
+    expect(instance).toBeDefined();
+    const xIndex = instance._getXIndex(xPoint2[0]);
+    expect(xIndex).toEqual('Fri Apr 03 2020 05:30:00 GMT+0530 (India Standard Time)');
```

Comment:
This test will break if run for other time zones. Put it under the skip category.

Reply:
Similarly all other unit tests involving date will break based on timezones of other machines where they are run.

Reply:
Moved respective tests into skipped block.